### PR TITLE
Athena Handler

### PR DESCRIPTION
**Pull request by rwatts3**
## Description

This pull request introduces the Athena Handler for MindsDB. The Athena Handler allows users to interact with Amazon Athena, an interactive query service for analyzing data in Amazon S3 using standard SQL. The handler was implemented using `boto3`, the AWS SDK for Python.

## Type of change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [x] **Test Location:** The testing can be done within the MindsDB integrations environment.
 - [x] **Verification Steps:**
   1. Set up the required AWS credentials and S3 bucket.
   2. Use the provided SQL commands in the README to create a connection to Athena.
   3. Run sample queries to verify data retrieval from Athena.
   4. Verify that the connection arguments are correctly passed and used by the handler.

## Additional Media:
- [x] I have attached a brief loom video or screenshots showcasing the new functionality or change.

<div>
    <a href="https://www.loom.com/share/349494ca3c4d4c728e37628e123fcccd">
      <p>Ryan Watts MindsDB - Athena Handler - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/349494ca3c4d4c728e37628e123fcccd">
      
    </a>
  </div>

## Checklist:

- [x] My code follows the style guidelines (PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [x] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/integrations/handlers/athena_handler/athena_handler.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        self.connect()
+        
+        try:
+            response = self.connection.start_query_execution(
+                QueryString=query,
+                QueryExecutionContext={
+                    'Database': self.connection_data['database'],
+                },
+                ResultConfiguration={
+                    'OutputLocation': self.connection_data['result_output_location'],
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/athena_handler/athena_handler.py:103]: I think this should be `results_output_location` ?

File: mindsdb/integrations/handlers/athena_handler/athena_handler.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+            query_execution_id (str): ID of the query to wait for
+        Returns:
+            str: Query execution status
+        """
+        while True:
+            response = self.connection.get_query_execution(QueryExecutionId=query_execution_id)
+            status = response['QueryExecution']['Status']['State']
+            if status in ['SUCCEEDED', 'FAILED', 'CANCELLED']:
+                return status
+            time.sleep(1)
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/athena_handler/athena_handler.py:195]: Do we need this? If so, maybe we can send a check_interval value as a param?

File: mindsdb/integrations/handlers/athena_handler/athena_handler.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+    def _parse_query_result(self, result) -> pd.DataFrame:
+        """
+        Parse the result of the Athena query into a DataFrame.
+        Args:
+            result: Result of the Athena query
+        Returns:
+            pd.DataFrame: Query result as a DataFrame
+        """
+        rows = result['ResultSet']['Rows']
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/athena_handler/athena_handler.py:205]: Maybe we can check if ResultSet exists first?

File: mindsdb/integrations/handlers/athena_handler/tests/test_athena_handler.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import unittest
+from mindsdb.integrations.handlers.athena_handler.athena_handler import AthenaHandler
+from mindsdb.api.executor.data_types.response_type import RESPONSE_TYPE
+
+
+class AthenaHandlerTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/athena_handler/tests/test_athena_handler.py:8]: Can we change these tests and use mocks? As a reference, you can check https://github.com/mindsdb/mindsdb/blob/main/tests/unit/handlers/test_mysql.py#L24

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]
Thank you @rwatts3 :raised_hands: I've added few minor comments

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by mindsdbadmin** [comment]
**CLA Assistant Lite bot:** <br/>Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you all sign our [Contributor License Agreement](https://github.com/mindsdb/mindsdb/blob/main/assets/contributions-agreement/individual-contributor.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.<br/>
   - - -
   I have read the CLA Document and I hereby sign the CLA
   - - -
   **0** out of **2** committers have signed the CLA.<br/>:x: @Ryan Watts<br/>:x: @rwatts3<br/>**Ryan Watts, rwatts3** seem not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You can retrigger this bot by commenting **recheck** in this Pull Request</sub>

**Comment by rwatts3** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by rwatts3** [comment]
recheck

**Comment by ZoranPandovski** [comment]
recheckcla

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]
Thanks @rwatts3 please just fix the flake8 warnings https://github.com/mindsdb/mindsdb/actions/runs/9427301817/job/26010434799?pr=9282#step:6:75

**Comment by rwatts3** [comment]
> Thanks @rwatts3 please just fix the flake8 warnings https://github.com/mindsdb/mindsdb/actions/runs/9427301817/job/26010434799?pr=9282#step:6:75

Done here b4e8963

**Comment by rwatts3** [comment]
Anything else you need from me on this PR?

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by ZoranPandovski** [comment]
Hey @rwatts3, thanks; this is ready for merging. I think locally you are logged in with one user and you don't have that git user linked to your GitHub account, that's why you can't sign the CLA. Can you please check that  [for more info check this link](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

**Comment by rwatts3** [comment]
recheck

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by ZoranPandovski** [comment]
recheck

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by ZoranPandovski** [comment]
recheckcla

**Pull request closed without merging**